policy/policy_handler/subscription.go

```diff
@@ -0,0 +1,36 @@
+package policy_handler
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/atomist-skills/go-skill"
+	"github.com/atomist-skills/go-skill/policy/data"
+	"olympos.io/encoding/edn"
+)
+
+func WithSubscription() Opt {
```

Comment:
I think most (all?) policies are going to want both the subcription and async handlers - do we want to add them by default?

Reply:
I did have the subscription one on by default, although since it matched the format I figured may as well make it an Opt. I see your comment above about making sync queries opt-in, I'll replace this with a default option that enables subscription payloads, and move sync queries to a new opt-in option.

Reply:
Yeah, just thinking about how to make it as easy as possible for policy authors. I think subscription should be default, and maybe async?

Local and sync should be opt-in I think.

Reply:
I think a good number of policies won't require any queries at all, so just subscription?

Reply:
Is there a cost to including async by default, even if policies don't use it?

Reply:
There shouldn't be any meaningful cost to having anything enabled by default. Just depends what things we want policy authors to be explicit about in their intents.

Reply:
I think that most policies should:

- get as much data via the subscription as possible
- if data is not available via subscriptions (such as package/vuln data) then it should be fetched by a single async query
- multiple async queries or sync queries should be a last resort

I'd say that having the first two as the default and the last ones opt-in probably makes sense.

Reply:
ok, done in c6c173958d5a8ffcc38f2c7048e1564590e25f13 and b330d7f437a91082b7285433f235e82c313cbb25